OpenGL/OpenGL/src/blocks/Block.h

```diff
@@ -0,0 +1,30 @@
+#ifndef BLOCK_H
+#define BLOCK_H
+
+#include "GL/glew.h"
+#include <vector>
+
+enum class Block_Type
+{
+    Air,
+    Grass,
+    Dirt,
+    Stone
+};
+
+class Block
+{
+public:
+    Block(Block_Type type);
+
+    virtual ~Block() { }
+
+    const Block_Type type;
+
+    virtual std::vector<GLfloat> getSideTxrCoords() const { return { 0 }; }
```

Comment:
I would prefer 'tex' instead of 'txr' as a short for texture.

```suggestion
    virtual std::vector<GLfloat> getSideTexCoords() const { return { 0 }; }
```

Reply:
That's a lot of changes to do lol
I'm working on it.
Do you have a VBO, Texture Loader.h/.cpp?

Reply:
sorry :/

> Do you have a VBO, Texture Loader.h/.cpp?

My previous project did not have any textures.
To handle the vbo, I used to have an abstract mesh class which generated the vbo from the vertex normals/colors etc. given by the implementation. Mesh also contained the model matrix. I then implemented Mesh in the ChunkMesh, PlayerMesh... classes. 

-> a ChunkMesh required a Chunk to generate the vertex data and then the parent Mesh class could create/manage the vbo.
-> a PlayerMesh essentially implements the TD3 to generate the player vertices.

So my rendering pipeline was:

- // update stuff
- activate shader
- activate camera
- foreach mesh in scene, mesh.draw()